### PR TITLE
Store the back end popup session bag under a different storage key

### PR DESCRIPTION
**Pull request by fritzmg**
This is an alternative approach to #7221 and #7269. It registers an additional session bag for the popup, under a different storage key, but under the same `contao_backend` name. This way you can still use the `contao_backend` session bag everywhere and don't have to worry about whether your back end module was opened in the popup or not. And you don't need an additional service to find the correct bag.

Caveat: this will not store the session state of the popup in the database (`tl_user.session`) anymore (same as in #7269 as far as I can see). So once you log out and then back in, all your filters etc. that were used in the popup only will be gone. But I think that's reasonable. However, if we don't want this, I can still change the `UserSessionListener` so that is stores both states (regular and popup).

### CODE REVIEW

**Review by Toflar — DISMISSED** [review]
I think I like this version the best because as an extension developer, I don't have to change anything at all 😄

**Review by Toflar — DISMISSED** [review]

File: core-bundle/config/services.yaml [hunk 1]
```diff
@@ -1056,6 +1056,13 @@ services:
         calls:
             - [setName, [contao_backend]]
 
+    contao.session.contao_backend_popup:
+        class: Contao\CoreBundle\Session\Attribute\ArrayAttributeBag
+        arguments:
+            - _contao_be_popup_attributes
+        calls:
+            - [setName, [contao_backend]]
+
```
Comment (leofeyer) [inline core-bundle/config/services.yaml]: It does not make sense to define the session bags as services if the session factory does not provide a way to dynamically add bags. For this reason, I would prefer to simplify the setup like I did in https://github.com/contao/contao/pull/7269/files#diff-c61014971921decbcd3006c93d1ab6ac2ba12fd11cdc69699cd597d361316117.

**Comment by leofeyer** [comment]
> Caveat: this will not store the session state of the popup in the database (`tl_user.session`) anymore

That‘s yet another problem I hadn‘t even thought about. So every time a user logs out, all open nodes are closed again and all filters and fieldsets are reset? Is that really a good UX?

**Comment by Toflar** [comment]
Yes, but only for things that are shown in the popup. I think that's even correct 🤔

File: core-bundle/config/services.yaml (hunk 1 above)
Comment (fritzmg) [reply core-bundle/config/services.yaml]: 👍  Also the bags are hardcoded everywhere anyway.

**Comment by fritzmg** [comment]
As an added note regarding the popup session state being stored or not stored in `tl_user.session`: if we want to change the `UserSessionListener` to also store the popup state, then we also need to increase the `blob` to a `mediumblob` I think, since it will then potentially have to store double the amount (and currently you might already run into an issue where the `blob` is insufficient in some rare cases).

**Comment by leofeyer** [comment]
See https://github.com/fritzmg/contao/pull/9

File: core-bundle/src/Session/SessionFactory.php [hunk 2]
```diff
@@ -12,25 +12,43 @@
...
-        readonly private SessionBagInterface $frontendBag,
+        readonly private RequestStack $requestStack,
+        readonly private ScopeMatcher $scopeMatcher,
     ) {
     }
 
     public function createSession(): SessionInterface
     {
         $session = $this->inner->createSession();
+        $request = $this->requestStack->getMainRequest();
```
Comment (leofeyer) [inline core-bundle/src/Session/SessionFactory.php]: Are we sure that this should not be `getCurrentRequest()`? 😄

Comment (fritzmg) [reply core-bundle/src/Session/SessionFactory.php]: The session is always attached to the main request. The `inner` service also uses `getMainRequest()` when creating the session.

Comment (fritzmg) [reply core-bundle/src/Session/SessionFactory.php]: I mean I suppose it's possible that a `?popup` _sub_-request can be made somewhere. But would that even trigger the session factory?

Comment (leofeyer) [reply core-bundle/src/Session/SessionFactory.php]: No, I don‘t think so.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @fritzmg.

**Pull request merged**